testnet/src/tests/neon_integrations.rs

```diff
@@ -0,0 +1,212 @@
+    eprintln!("== REGTEST BLOCK MINED == ");
+    // give the run loop some time to figure things out!
+    thread::sleep(std::time::Duration::from_millis(gap_ms));
+
+    // first block will hold our VRF registration
+    btc_regtest_controller.build_next_block(1);
+    eprintln!("== REGTEST BLOCK MINED == ");
+    // give the run loop some time to figure things out!
+    thread::sleep(std::time::Duration::from_millis(gap_ms));
+
+    // second block will be the first mined Stacks block
+    btc_regtest_controller.build_next_block(1);
+    eprintln!("== REGTEST BLOCK MINED == ");
+    thread::sleep(std::time::Duration::from_millis(gap_ms));
+
+    // let's query the miner's account nonce:
+
+    eprintln!("Miner account: {}", miner_account);
+
+    let path = format!("{}/v2/accounts/{}?proof=0",
+                       &http_origin, &miner_account);
+    eprintln!("Test: GET {}", path);
+    let res = client.get(&path).send().unwrap().json::<AccountEntryResponse>().unwrap();
+    assert_eq!(u128::from_str_radix(&res.balance[2..], 16).unwrap(), 0);
+    assert_eq!(res.nonce, 1);
+
+    // and our spender
+
+    let path = format!("{}/v2/accounts/{}?proof=0",
+                       &http_origin, &spender_addr);
+    let res = client.get(&path).send().unwrap().json::<AccountEntryResponse>().unwrap();
+    assert_eq!(u128::from_str_radix(&res.balance[2..], 16).unwrap(), 100300);
+    assert_eq!(res.nonce, 0);
+
+    // okay, let's push a transaction that is marked microblock only!
+    let recipient = StacksAddress::from_string(ADDR_4).unwrap();
+    let tx = make_stacks_transfer_mblock_only(&spender_sk, 0, 1000, &recipient.into(), 1000);
+
+    let path = format!("{}/v2/transactions", &http_origin);
+    let res: String = client.post(&path)
+        .header("Content-Type", "application/octet-stream")
+        .body(tx.clone())
+        .send()
+        .unwrap()
+        .json()
+        .unwrap();
+
+    assert_eq!(res, StacksTransaction::consensus_deserialize(&mut &tx[..]).unwrap().txid().to_string());
+
+    // now let's mine a couple blocks, and then check the sender's nonce.
+    // this one wakes up our node, so that it'll mine a microblock _and_ an anchor block.
+    btc_regtest_controller.build_next_block(1);
+    eprintln!("== REGTEST BLOCK MINED == ");
+    thread::sleep(std::time::Duration::from_millis(gap_ms));
+    // this one will contain the sortition from above anchor block,
+    //    which *should* have also confirmed the microblock.
+    btc_regtest_controller.build_next_block(1);
+    eprintln!("== REGTEST BLOCK MINED == ");
+    thread::sleep(std::time::Duration::from_millis(gap_ms));
+
+
+    let path = format!("{}/v2/accounts/{}?proof=0",
+                       &http_origin, &spender_addr);
+    let res = client.get(&path).send().unwrap().json::<AccountEntryResponse>().unwrap();
+    eprintln!("{:#?}", res);
+    assert_eq!(u128::from_str_radix(&res.balance[2..], 16).unwrap(), 98300);
+    assert_eq!(res.nonce, 1);
```

Comment:
Can you expand this test so that it mines a sequence of anchored-block/microblock pairs?  i.e. to confirm that the miner can pick up the microblock and build on it?

Reply:
That's already what is tested here -- the RPC interface is returning data from the chainstate at the block which _confirms_ the microblock (and for the transaction to have been included, it must have been via a microblock) -- I can add another account info lookup for the miner's nonce (which would test the assertion that blocks are continuing to mine).